pilot/pkg/xds/ads.go

```diff
@@ -158,25 +158,45 @@ func isExpectedGRPCError(err error) bool {
 	return false
 }
 
-func receiveThread(con *Connection, reqChannel chan *discovery.DiscoveryRequest, errP *error) {
+func (s *DiscoveryServer) receiveThread(con *Connection, reqChannel chan *discovery.DiscoveryRequest, errP *error) {
 	defer close(reqChannel) // indicates close of the remote side.
+	firstReq := true
 	for {
 		req, err := con.stream.Recv()
 		con.mu.RLock()
 		cid := con.ConID
 		con.mu.RUnlock()
 		if err != nil {
+			cid := con.ConID
 			if isExpectedGRPCError(err) {
-				con.mu.RLock()
```

Comment:
was this here for PeerAddr? is there still a race there? I guess probably no because line 176 doesn't have it